### PR TITLE
[KB-255] Reorganize background

### PR DESCRIPTION
**Pull request by Ivikhostrup**
Reading guide:
* What do you think of the reorg? Is it what we'd like it to be?
* I've written an introduction to the background section, the preprocessing section, for ensemble learning and linear & regularization models.
    * However, I feel like an introduction to linear & regularization models seems a little.. Redundant in a way? I dont have much to say. Perhaps you have some ideas.

### CODE REVIEW

File: report_thesis/src/sections/background.tex [hunk 1]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
 By constructing a kernel matrix $\mathbf{K}$ using these pairwise similarities, \gls{kernel-pca} can perform eigenvalue decomposition to obtain the principal components in the feature space, similar to regular \gls{pca} as described in Section~\ref{subsec:pca}.
 However, in \gls{kernel-pca}, the eigenvalue decomposition is performed on the kernel matrix $\mathbf{K}$ rather than the covariance matrix $\mathbf{C}$.
 
-\subsection{Overview of Core Models}
-In this section, we provide an overview and definitions of \gls{pls}, \gls{svr}, \gls{etr}, \gls{gbr}, and \gls{xgboost}.
-These models form the basis of the final architecture of our proposed pipeline, detailed further in Section~\ref{sec:methodology}.
+\subsection{Ensemble Learning Models}
+In this section we introduce the concept of ensemble learning and decision trees, as they are fundamental aspects of the ensemble learning models we discuss.
+Following this we outline \gls{etr}, \gls{gbr}, \gls{ngboost}, and \gls{xgboost}.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:188]:
```suggestion
In this section we introduce the concept of ensemble learning and decision trees based on \citet{James2023AnIS}, as they are fundamental aspects of the ensemble learning models we discuss.
Following this, we outline \gls{etr}, \gls{gbr}, \gls{ngboost}, and \gls{xgboost}.
```

File: report_thesis/src/sections/background.tex [hunk 2]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+
+The parameters of the model are then updated using the output from the weak learner.
+This update process incorporates a learning rate to control the step size, ensuring that the model makes gradual improvements rather than drastic changes.
+
+Using the newly updated parameters, the model recalculates its predictions, refining the probability distribution of the target variable. 
+This iterative process of computing predictions, calculating gradients, fitting weak learners, and updating parameters continues for a predetermined number of iterations or until the model's performance converges.
+
+\subsubsection{XGBoost}
 
+\subsection{Linear and Regularization Models}
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:290]: We don't actually use any regularization models. I'd think of Elastic Net Regression, Ridge Regression, or LASSO Regression in that context, which we don't employ. They are, however, utilized in other LIBS-based projects.

File: report_thesis/src/sections/background.tex [hunk 3]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+The repetitive process of fitting a weak learner to predict the residuals and using its predictions to update the model gives us the final model:
+$$
+    \hat{f}(\mathbf{x}) = \sum_{b=1}^{B} \lambda \hat{f}_b(\mathbf{x})
+$$
+In the context of regression, gradient boosting aims to minimize the difference between the predicted values and the actual target values by fitting successive trees to the residuals.
+To minimize errors, gradient descent is used to iteratively update model parameters in the direction of the negative gradient of the loss function, thereby following the path of steepest descent~\cite{gradientLossFunction}.
+
+\subsubsection{Natural Gradient Boosting (NGBoost)}
+Having introduced \gls{gbr}, we now give an overview of \gls{ngboost} based on \citet{duan_ngboost_2020}.
+
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:254]: Implied
```suggestion
```

File: report_thesis/src/sections/background.tex [hunk 4]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+While \gls{rf} uses bootstrap sampling and selects the best split from a random subset of features to create a set of diverse samples, \gls{etr} instead selects split points randomly within the chosen features, introducing additional randomness.
+This process results in even greater variability among the trees, aiming to reduce overfitting and improve the model's robustness.
+As a tradeoff, \gls{etr} is less interpretable than a single decision tree, as the added randomness can introduce more bias than \gls{rf}.
+However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
+
+\subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
+In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
+\gls{gbr} is an ensemble learning method that sequentially construct multiple decision trees, each one correcting the errors of the previous one, using gradient descent and boosting techniques.
+Each tree is small, with few terminal nodes, preventing large adjustments based on a single tree's predictions.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:231]: Use `leaf` elsewhere
```suggestion
Each tree is small, with few leaf nodes, preventing large adjustments based on a single tree's predictions.
```

File: report_thesis/src/sections/background.tex [hunk 5]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+
+\gls{etr} extends the \gls{rf} model by introducing additional randomness in the tree-building process, specifically through random feature selection and random split points.
+While \gls{rf} uses bootstrap sampling and selects the best split from a random subset of features to create a set of diverse samples, \gls{etr} instead selects split points randomly within the chosen features, introducing additional randomness.
+This process results in even greater variability among the trees, aiming to reduce overfitting and improve the model's robustness.
+As a tradeoff, \gls{etr} is less interpretable than a single decision tree, as the added randomness can introduce more bias than \gls{rf}.
+However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
+
+\subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
+In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
+\gls{gbr} is an ensemble learning method that sequentially construct multiple decision trees, each one correcting the errors of the previous one, using gradient descent and boosting techniques.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:230]: Boosting is literally what you describe: sequentially constructing trees (one at a time...), each one correcting the errors of the previous one. That's the boosting technique it uses - there aren't more.

The innovation with gradient boosting is to use gradient descent to minimize a loss function (which could be for regression, classification, etc., as long as the function is continuous and differentiable).
We start with some initial model. We compute the gradients of the loss function, w.r.t. the prediction of the current ensemble (just the initial model atm.). Then we fit a _new_ learner to the gradients. Then we add that learner to the ensemble, weighing it by some factor (i.e. a learning rate). And we repeat that. That's the 'gradient descent' process here.

File: report_thesis/src/sections/background.tex [hunk 6]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+$$
+    r_i^{(b)} = y_i - \hat{f}^{(b-1)}(\mathbf{x}_i),
+$$
+Each tree is then trained on these updated residuals.
+The repetitive process of fitting a weak learner to predict the residuals and using its predictions to update the model gives us the final model:
+$$
+    \hat{f}(\mathbf{x}) = \sum_{b=1}^{B} \lambda \hat{f}_b(\mathbf{x})
+$$
+In the context of regression, gradient boosting aims to minimize the difference between the predicted values and the actual target values by fitting successive trees to the residuals.
+To minimize errors, gradient descent is used to iteratively update model parameters in the direction of the negative gradient of the loss function, thereby following the path of steepest descent~\cite{gradientLossFunction}.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:250]: In the context of regression, gradient boosting aims to minimize prediction errors by fitting successive decision trees to the residuals of previous predictions. Each new tree corrects errors made by previous trees. Gradient descent is used to optimize the model parameters by updating them in the direction of the negative gradient of the loss function, helping the model minimize errors efficiently.

File: report_thesis/src/sections/background.tex [hunk 7]
```diff
@@ -178,16 +183,118 @@ \subsubsection{Kernel PCA}
...
+where $f_m(x)$ is the prediction of the $m$-th tree, and $M$ is the total number of trees.
+
+\gls{etr} extends the \gls{rf} model by introducing additional randomness in the tree-building process, specifically through random feature selection and random split points.
+While \gls{rf} uses bootstrap sampling and selects the best split from a random subset of features to create a set of diverse samples, \gls{etr} instead selects split points randomly within the chosen features, introducing additional randomness.
+This process results in even greater variability among the trees, aiming to reduce overfitting and improve the model's robustness.
+As a tradeoff, \gls{etr} is less interpretable than a single decision tree, as the added randomness can introduce more bias than \gls{rf}.
+However, it often achieves better generalization performance, especially in high-dimensional or noisy datasets.
+
+\subsubsection{Gradient Boosting Regression (GBR)}\label{sec:gradientboost}
+In this section we introduce \gls{gbr} primarily based on \citet{James2023AnIS}.
```
Comment (chhoumann) [inline report_thesis/src/sections/background.tex:229]: Ahh, I see, this book doesn't explain Gradient Boosting. It only explains boosting. My bad for not catching on earlier. We don't even explain GB here, only boosting. This entire section needs to get updated. It's great that we explain boosting, but let's do that, then explain GM (or Gradient Boosting Machines), and finally explain GBR, which just uses a loss function suitable for regression.

**Review by chhoumann — COMMENTED** [review]
Restructure is good. Found some general issues with the section we'll need to address.

File: report_thesis/src/sections/background.tex [hunk 8]
```diff
@@ -1,5 +1,11 @@
 \section{Background}\label{sec:background}
+In this section, we provide an overview of the preprocessing techniques and machine learning models used in our proposed pipeline. 
+We outline the various normalization techniques and dimensionality reduction methods, followed the ensemble learning and linear and regularization models used.
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply report_thesis/src/sections/background.tex:3]: Keep the regularized models. Will write.

File: report_thesis/src/sections/background.tex (hunk 2 above)
Comment (chhoumann) [reply report_thesis/src/sections/background.tex:290]: We do now

File: report_thesis/src/sections/background.tex (hunk 7 above)
Comment (Ivikhostrup) [reply report_thesis/src/sections/background.tex:230]: I resolve these because they are handled by another PR.

File: report_thesis/src/sections/background.tex (hunk 5 above)
Comment (Ivikhostrup) [reply report_thesis/src/sections/background.tex:231]: I resolve these because they are handled by another PR.

File: report_thesis/src/sections/background.tex (hunk 4 above)
Comment (Ivikhostrup) [reply report_thesis/src/sections/background.tex:232]: I resolve these because they are handled by another PR.

File: report_thesis/src/sections/background.tex (hunk 6 above)
Comment (Ivikhostrup) [reply report_thesis/src/sections/background.tex:251]: I resolve these because they are handled by another PR.

**Review by chhoumann — APPROVED** [review]
